src/constraints/load_balance.jl

```diff
@@ -135,34 +135,32 @@ function add_thermal_load_constraints(m, p; _n="")
     else    
         if !isempty(p.techs.heating)
             
-            # if !isempty(p.SteamTurbineTechs)
-            #     @constraint(m, HotThermalLoadCon[ts in p.time_steps],
-            #             # sum(m[Symbol("dvThermalProduction"*_n)][t,ts] - m[:dvThermalToSteamTurbine][t,ts] for t in p.CHPTechs) +
-            #             # sum(m[Symbol("dvThermalProduction"*_n)][t,ts] for t in p.SteamTurbineTechs) +
-            #             sum(p.production_factor[t,ts] * (m[Symbol("dvThermalProduction"*_n)][t,ts] - m[:dvThermalToSteamTurbine][t,ts]) for t in p.techs.boiler)
-            #             # + sum(p.GHPHeatingThermalServed[g,ts] * m[Symbol("binGHP"*_n)][g] for g in p.ghp_options)
-            #             # + sum(m[Symbol("dvDischargeFromStorage"*_n)][b,ts] for b in p.s.storage.types.hot) 
-            #             ==
-            #             p.HeatingLoad[ts] * p.s.existing_boiler.efficiency
-            #             # + sum(m[:dvProductionToWaste][t,ts] for t in p.CHPTechs) +
-            #             # sum(m[:dvProductionToStorage][b,t,ts] for b in p.s.storage.types.hot, t in p.techs.heating) +
-            #             # sum(m[Symbol("dvThermalProduction"*_n)][t,ts] / p.thermal_cop[t] for t in p.techs.absorption_chiller) -
-            #             # sum(p.heating_thermal_load_reduction_with_ghp_kw[g,ts] * m[Symbol("binGHP"*_n)][g] for g in p.ghp_options) 
-            #     )
-            # else
+            if !isempty(p.techs.steam_turbine)
+                @constraint(m, [ts in p.time_steps],
+                        sum(m[Symbol("dvThermalProduction"*_n)][t,ts] - m[Symbol("dvThermalToSteamTurbine"*_n)][t,ts] for t in p.techs.all_for_steam_turbine) +
+                        sum(m[Symbol("dvThermalProduction"*_n)][t,ts] for t in p.techs.steam_turbine) +
+                        + sum(m[Symbol("dvDischargeFromStorage"*_n)][b,ts] for b in p.s.storage.types.hot)
+                        + sum(p.ghp_heating_thermal_load_served_kw[g,ts] * m[Symbol("binGHP"*_n)][g] for g in p.ghp_options)
+                        ==
+                        (p.s.dhw_load.loads_kw[ts] + p.s.space_heating_load.loads_kw[ts])
+                        + sum(m[Symbol("dvProductionToWaste"*_n)][t,ts] for t in p.techs.chp) +
+                        sum(m[Symbol("dvProductionToStorage"*_n)][b,t,ts] for b in p.s.storage.types.hot, t in union(p.techs.heating, p.techs.chp))  +
+                        sum(m[Symbol("dvThermalProduction"*_n)][t,ts] / p.thermal_cop[t] for t in p.techs.absorption_chiller) -
+                        sum(p.heating_thermal_load_reduction_with_ghp_kw[g,ts] * m[Symbol("binGHP"*_n)][g] for g in p.ghp_options)
```

Comment:
Will do

Reply:
I prefer this format too as I believe this is as readable as it gets for really large constraints (which we have a few of!).  Thanks for this!  I'm adding new issue https://github.com/NREL/REopt.jl/issues/132 to address this update sometime after v1.0 when are far fewer live branches (to avoid lots of merge conflicts).